### PR TITLE
fix #2108

### LINKED ISSUE #2108: [BUG] order of cov in ADVI init is incorrect
While updating a notebook of mine, I found that two mixed model used to ran quite fast with good trace now becomes quite slow (10x slower) and the estimation is off (see the trace). Since the advi is one of the main change (I ran the same notebook around 45 days ago), it might be the main reason.
I put together a notebook [gist](https://gist.github.com/junpenglao/272909957daa3a26fb7e5a1d9dcdff66). I am not sure if this is a common issue - but it would be good if @ferrine you can have a closer look at this.

I dont think the estimation from the old ADVI was very accurate (in fact looking at the trace they are not),  the problem is then the `cov` from the ADVI. It seems to be a bug that the order of the cov is different than before, but we are pluging it into `pm.NUTS` the same way.

### PR DESCRIPTION
**Pull request by junpenglao**

### CODE REVIEW

**Comment by junpenglao** [comment]
fix #2108 @twiecki @ferrine

**Comment by ferrine** [comment]
What is wrong with order? Is it the best place for fix?

**Comment by junpenglao** [comment]
@ferrine basically, the problem is that the somehow the bijection.map in approximation class is different than the `model.bijection.map`. So in the 
https://github.com/pymc-devs/pymc3/blob/master/pymc3/sampling.py#L588-L589, the `cov` has a different order compare to `start`.
As a result, the pm.NUTS get the wrong scaling.

In the old advi init, it does `cov = np.power(model.dict_to_array(v_params.stds), 2)` to make sure the `cov` has the right order.
https://github.com/pymc-devs/pymc3/blob/bb25d7bb43aa0684fdbd7628546be044c873b27b/pymc3/sampling.py#L474

**Comment by twiecki** [comment]
Seems like there's a underlying issue which we should fix instead.

**Comment by junpenglao** [comment]
@twiecki I agree - something must happen during `pm.fit`, which somehow operate on `model`.
However, it might be useful to have this fix in place as a fail-safe

File: pymc3/sampling.py [hunk 1]
```diff
@@ -573,7 +573,15 @@ def init_nuts(init='ADVI', njobs=1, n_init=500000, model=None,
             progressbar=progressbar
         )  # type: pm.MeanField
         start = approx.sample(draws=njobs)
-        cov = approx.cov.eval()
```
Comment (ferrine) [inline pymc3/sampling.py:576]: that's more pretty
```
stds = mf.gbij.rmap((tt.diag(mf.cov).eval()))
cov = model.dict_to_array(stds)
```

Comment (junpenglao) [reply pymc3/sampling.py:576]: thanks! I will go with this. 
If I understand correctly the posterior cov of `advi` is a diagonal matrix right?

Comment (ferrine) [reply pymc3/sampling.py:576]: Yes

**Comment by twiecki** [comment]
```
>           stds = approx.gbij.rmap((tt.diag(approx.cov).eval()))
E           NameError: global name 'tt' is not defined
```

**Comment by junpenglao** [comment]
Oops, thanks @twiecki

**Pull request merged**